index.js

```diff
@@ -42,13 +52,44 @@ exports.register = (server, options, next) => {
                 if(!route.config.validate) {
                     route.config.validate = {};
                 }
-                
+
                 route.config.validate.headers = Joi.object({
                         Authorization: Joi.string().description('JWT token')
                     }).unknown();
             }
 
             server.route(route);
```

Comment:
We are adding the route to server before modifying it. That line should be after the `if` block that follows.